assets/global.js

```diff
@@ -771,20 +780,34 @@ class SlideshowComponent extends SliderComponent {
 
   onButtonClick(event) {
     super.onButtonClick(event);
+
     const isFirstSlide = this.currentPage === 1;
     const isLastSlide = this.currentPage === this.sliderItemsToShow.length;
 
-    if (!isFirstSlide && !isLastSlide) return;
+    if (!isFirstSlide && !isLastSlide) {
+      this.appleAnimationToAnnouncementBar(event.currentTarget.name);
+      return;
+    }
 
     if (isFirstSlide && event.currentTarget.name === 'previous') {
       this.slideScrollPosition =
         this.slider.scrollLeft + this.sliderFirstItemNode.clientWidth * this.sliderItemsToShow.length;
     } else if (isLastSlide && event.currentTarget.name === 'next') {
       this.slideScrollPosition = 0;
     }
-    this.slider.scrollTo({
-      left: this.slideScrollPosition,
-    });
+
+    this.setSlidePosition(this.slideScrollPosition);
+
+    this.appleAnimationToAnnouncementBar(event.currentTarget.name);
+  }
```

Comment:
One thing I noticed is that all the logic is added to be tackled onButtonClick when it should also be running on scroll/swipe 😬 

If you use it on mobile and swipe instead of you will see that it doesn't apply all the changes necessary.

So I think you could either look into: 
- applying some of the `announcement-bar-slider--fade-***` class in liquid by default. Since you know which slide will be showing on page load and which ones won't. 
- look at applying the same transitions in the `update()` function where it's checking what is the current page and what's the previous page. If you apply the logic you added to `onButtonClick` there it should work in all cases. Tho it might be triggered too late compared to when the button is clicked 🤔  


From what I'm seeing it's working ok on page load. No animation is applied but at least all slides are being shown. If you click on the previous or next button though, and then swipe later, the slides are hidden. Ideally they would work whatever the method you're using to go to the next slide is. 

https://github.com/Shopify/dawn/assets/28613840/9913deb6-a047-47a5-90a6-616f1291effd

Reply:
Thanks for flagging this Ludo. What if we just restrict animation if you swipe/scroll it?

Honestly, in my opinion, when you swipe it looks way more better with the original effect rather than the current one? Because when you swipe you want it to be swiped and not fade out. Thoughts?